### PR TITLE
New Measure tool!

### PR DESCRIPTION
**Pull request by theduckylittle**
1. Added new "measure:XXX" type interactions which will
   start an interaction for drawing.
2. Added new actions/reducer support for setting and clearing
   a measured feature.
3. Updated the drawing example to include measure tools:
   a. The measure tools themselves are now availabe in the tool dropdown.
   b. Created a new MeasureTable component in the drawing example
      that renders the results of a measure tool.

REFS: SDK-525

### CODE REVIEW

File: __tests__/components/map-drawing.test.jsx [hunk 1]
```diff
@@ -219,5 +222,110 @@ if (HAS_CANVAS) {
...
+      // set the point measure
+      store.dispatch(DrawingActions.startMeasure(INTERACTIONS.measure_point));
+      // get the measure interaction
+      const interactions = ol_map.getInteractions();
+      const measure = interactions.item(interactions.getLength() - 1);
+
+      // create a dummy OL feature with a Point for measuring.
+      const sketch_feature = new Feature();
+      const sketch_geometry = new Point([100, 100]);
+      sketch_feature.setGeometry(sketch_geometry);
```
Comment (bartvde) [inline __tests__/components/map-drawing.test.jsx]: minor but you can do this in ol with 1 line less:

```
const sketch_geometry = new Point([100, 100]);
const sketch_feature = new Feature(sketch_geometry);
```

File: __tests__/components/map-drawing.test.jsx [hunk 2]
```diff
@@ -219,5 +222,110 @@ if (HAS_CANVAS) {
...
+          coordinates: [0, 0],
+        },
+      });
+    });
+
+    it('measures a line', () => {
+      const sdk_map = wrapper.instance().getWrappedInstance();
+      const ol_map = sdk_map.map;
+
+      // set the point measure
```
Comment (bartvde) [inline __tests__/components/map-drawing.test.jsx]: point -> line

File: __tests__/components/map-drawing.test.jsx [hunk 3]
```diff
@@ -219,5 +222,110 @@ if (HAS_CANVAS) {
...
+      store.dispatch(DrawingActions.startMeasure(INTERACTIONS.measure_line));
+      // get the measure interaction
+      const interactions = ol_map.getInteractions();
+      const measure = interactions.item(interactions.getLength() - 1);
+
+      // create a dummy OL feature with a Point for measuring.
```
Comment (bartvde) [inline __tests__/components/map-drawing.test.jsx]: Point -> LineString

File: __tests__/components/map-drawing.test.jsx [hunk 4]
```diff
@@ -219,5 +222,110 @@ if (HAS_CANVAS) {
...
+          coordinates: coords,
+        },
+      });
+    });
+
+    it('measures a polygon', () => {
+      const sdk_map = wrapper.instance().getWrappedInstance();
+      const ol_map = sdk_map.map;
+
+      // set the point measure
```
Comment (bartvde) [inline __tests__/components/map-drawing.test.jsx]: point -> polygon

File: __tests__/components/map-drawing.test.jsx [hunk 5]
```diff
@@ -219,5 +222,110 @@ if (HAS_CANVAS) {
...
+      store.dispatch(DrawingActions.startMeasure(INTERACTIONS.measure_polygon));
+      // get the measure interaction
+      const interactions = ol_map.getInteractions();
+      const measure = interactions.item(interactions.getLength() - 1);
+
+      // create a dummy OL feature with a Point for measuring.
```
Comment (bartvde) [inline __tests__/components/map-drawing.test.jsx]: Point -> Polygon

File: src/components/map.jsx [hunk 6]
```diff
@@ -783,6 +790,34 @@ export class Map extends React.Component {
...
+      measure.on('drawstart', (evt) => {
+        const geom = evt.feature.getGeometry();
+        const proj = this.map.getView().getProjection();
+
+        measure_listener = geom.on('change', (geomEvt) => {
+          this.props.setMeasureGeometry(geomEvt.target, proj);
+        });
+
+        this.props.setMeasureGeometry(geom, proj);
+      });
```
Comment (bartvde) [inline src/components/map.jsx]: this probably works fine wrt scope, but an explicit ```measure.on('drawstart', (evt) => {}, this)``` would make it more obvious to the reader?

File: __tests__/components/map-drawing.test.jsx (hunk 1 above)
Comment (theduckylittle) [inline __tests__/components/map-drawing.test.jsx]: Fixed in the appropriate places.

File: src/components/map.jsx (hunk 6 above)
Comment (theduckylittle) [inline src/components/map.jsx]: So "Fat arrow" function's intent is that they do not change scope. Any `() => {}` already means that it stays bound to `this`.

**Pull request merged**